### ISSUE TITLE
TypeError array_map() while saving HasManyRepeater field containing FileUpload Field on EditPage

### ISSUE THREAD

**Issue by zzkevinlim** [laravel-filament/filament#1396]
### Package

filament/filament

### Package Version

v2.9,15

### Laravel Version

v8.81.0

### Livewire Version

v2.10.1

### PHP Version

PHP 8.0.14

### Bug description

TypeError array_map() while saving HasManyRepeater field containing FileUpload Field on EditPage.

However, the files do get uploaded and the data gets inserted into db. 

### Steps to reproduce

[https://i.imgur.com/CqJrrUs.mp4](https://i.imgur.com/CqJrrUs.mp4)

### Relevant log output

```
TypeError
array_map(): Argument #2 ($array) must be of type array, string given
```
```
Illuminate\Contracts\Filesystem\FileNotFoundException
File not found at path: livewire-tmp/ea8CbpkDXM5LoeyBDnW4xhS7txgJcF-metac2FtcGxlLnBkZg==-.pdf
```

**Comment by nerdroid23** [comment]
also having said issue, files get uploaded and data is saved in db.

**Comment by romaldyminaya** [comment]
Hi,

I have the same exact issue :(

**Comment by romaldyminaya** [comment]
To resolve this issue just go  to the file:

`filament\forms\src\Components\BaseFileUpload.php`

And in the line 413 paste the code above:

```
        if (!is_array($this->getState())) {
            $this->state([$this->getState()]);
        }
```

So the entire method should look like:

```
    public function saveUploadedFiles(): void
    {
        if (blank($this->getState())) {
            $this->state([]);

            return;
        }

        if (!is_array($this->getState())) {
            $this->state([$this->getState()]);
        }

        $state = array_map(function (TemporaryUploadedFile | string $file) {

            if (!$file instanceof TemporaryUploadedFile) {
                return $file;
            }


            $callback = $this->saveUploadedFileUsing;

            if (!$callback) {
                $file->delete();

                return $file;
            }

            $storedFile = $this->evaluate($callback, [
                'file' => $file,
            ]);

            $file->delete();

            return $storedFile;
        }, $this->getState());

        if ($this->canReorder && ($callback = $this->reorderUploadedFilesUsing)) {
            $state = $this->evaluate($callback, [
                'state' => $state,
            ]);
        }

        $this->state($state);
    }
```

Please @danharrin look this...

**Comment by danharrin** [comment]
Fixed by #1529.
